packages/thirdweb/src/react/web/ui/prebuilt/NFT/index.tsx

```diff
@@ -0,0 +1,83 @@
+"use client";
+
+import { createContext, useContext } from "react";
+import type { ThirdwebContract } from "../../../../../contract/contract.js";
+import { tokenURI } from "../../../../../extensions/erc721/__generated__/IERC721A/read/tokenURI.js";
+import { uri } from "../../../../../extensions/erc1155/__generated__/IERC1155/read/uri.js";
+import { useReadContract } from "../../../../../react/core/hooks/contract/useReadContract.js";
+import type { NFTMetadata } from "../../../../../utils/nft/parseNft.js";
+import { MediaRenderer } from "../../MediaRenderer/MediaRenderer.js";
+import type { MediaRendererProps } from "../../MediaRenderer/types.js";
+
+export type NFTProviderProps = {
+  contract: ThirdwebContract;
+  tokenId: bigint;
+};
+
+export const NFTProviderContext = /* @__PURE__ */ createContext<
+  NFTProviderProps | undefined
+>(undefined);
+
+export function NFT(props: React.PropsWithChildren<NFTProviderProps>) {
+  return (
+    <NFTProviderContext.Provider value={props}>
+      {props.children}
+    </NFTProviderContext.Provider>
+  );
+}
+
+/**
+ * @internal
+ */
+function useNFTContext() {
+  const ctx = useContext(NFTProviderContext);
+  return ctx;
+}
+
+export type NFTMediaProps = MediaRendererProps & NFTProviderProps;
+
+NFT.Media = (props: NFTProviderProps) => {
+  const context = useNFTContext();
+  const contract = props.contract || context;
+  const mediaQuery = useReadContract(getNFTMedia, {
+    contract,
+    tokenId: props.tokenId,
+  });
+  const animation_url = mediaQuery.data?.animation_url;
+  const image = mediaQuery.data?.image;
+  return (
+    <MediaRenderer
+      client={contract.client}
+      src={animation_url || image}
+      poster={image}
+      {...props}
+    />
```

Comment:
need a way to pass `className` and `style` prop through to media renderer

Reply:
and u probably will want to re-write the media render so that it has 0 built in styling etc for this (can copy paste and have a different one for this potentially?)

Reply:
The props for NFT.Media is:
```
export type NFTMediaProps = MediaRendererProps & NFTProviderProps;
```

which mean users can pass the styles + classes to it, as if they are using MediaRender right

Reply:
(that's why I put `{...props}` there btw)